### PR TITLE
feat(managers): implement algorithm for peer discovery in SessionManager

### LINKED ISSUE #90: Implement algorithm for peer discovery in SessionsManager
#### Context
- `SessionsManager` should be the component in charge of periodically asking for potential peer addresses using any of the currently open outbound peer sessions.
- The period for running this routine must be read from the configuration manager.
- The app configuration file must include an entry for setting the value of the period.
- For the scope of `testnet-1`, we will not be asking every peer for their own peer addresses, but only one of them, selected randomly. This is easy thanks to #87.
- The obtained peers must be inserted into the state of `PeersManager`.
- While this describes implementation of the `SessionManagers` side of the protocol, the `Session` actor side is in #96.

#### Actionables
- [ ] Decide the name of the peer discovery period variable
- [ ] Make config manager load the peer discovery period variable from the configuration file
- [ ] Update documentation to reflect existence, purpose and usage of the peer discovery period  variable
- [ ] Make `SessionsManager` ask `ConfigManager` for the value of the period upon start.
- [ ] Make `SessionsManager` schedule a periodic call to itself with `Anycast(GetPeers)` (see #87).
- [ ] Add documentation for this algorithm in the project-level documentation (mkdocs), under `architecture` as this is not protocol relevant.

### PR DESCRIPTION
**Pull request by lrubiorod**
Define a new config parameter named discovery_peers_period.
Implement a function on Session Manager to call to itself with Anycast(GetPeers) (see #87).
Add documentation to SessionsManager and Sessions

Close #90

### CODE REVIEW

File: core/src/actors/sessions_manager.rs [hunk 1]
```diff
@@ -150,6 +165,9 @@ impl Actor for SessionsManager {
 
             // We'll start the bootstrap peers process on sessions manager start
             act.bootstrap_peers(ctx, bootstrap_peers_period);
+
+            // We'll start the discovery peers process on sessions manager start
```
Comment (aesedepece) [inline core/src/actors/sessions_manager.rs]:
```suggestion
            // We'll start the peers discovery process upon SessionsManager's start
```

File: core/src/actors/sessions_manager.rs [hunk 2]
```diff
@@ -150,6 +165,9 @@ impl Actor for SessionsManager {
 
             // We'll start the bootstrap peers process on sessions manager start
```
Comment (aesedepece) [inline core/src/actors/sessions_manager.rs]:
```suggestion
            // We'll start the peers bootstrapping process upon SessionsManager's start
```

File: docs/architecture/managers/sessions-manager.md [hunk 3]
```diff
@@ -47,11 +47,12 @@ System::current().registry().set(sessions_manager_addr);
...
-| Unregister    | `SocketAddr, SessionType`                 | `SessionsResult<()>`  | Request to unregister a session           |
-| Update        | `SocketAddr, SessionType, SessionStatus`  | `SessionsResult<()>`  | Request to update the status of a session |
+| Message       | Input type                                | Output type           | Description                                    |
+|---------------|-------------------------------------------|-----------------------|------------------------------------------------|
+| Register      | `SocketAddr, Addr<Session>, SessionType`  | `SessionsResult<()>`  | Request to register a new session              |
+| Unregister    | `SocketAddr, SessionType`                 | `SessionsResult<()>`  | Request to unregister a session                |
+| Update        | `SocketAddr, SessionType, SessionStatus`  | `SessionsResult<()>`  | Request to update the status of a session      |
+| Anycast<T>    | `T`                                       | `()`                  | Request to send a T message to a random Session|
```
Comment (aesedepece) [inline docs/architecture/managers/sessions-manager.md]: Please put all messages between backticks (``)

File: docs/architecture/managers/sessions-manager.md [hunk 4]
```diff
@@ -103,17 +104,21 @@ session_manager_addr
     })
     .wait(ctx);
 ```
+#### Anycast<T>
+The handling of this messages is basically just calling the method `get_random_anycast_session` from the
```
Comment (aesedepece) [inline docs/architecture/managers/sessions-manager.md]:
```suggestion
The handler for `Anycast<T>` messages is basically just calling the method `get_random_anycast_session` from the
```

File: docs/architecture/managers/sessions-manager.md [hunk 5]
```diff
@@ -103,17 +104,21 @@ session_manager_addr
...
+[`Sessions`][sessions] library to obtain a random Session and send a `T` message.
```
Comment (aesedepece) [inline docs/architecture/managers/sessions-manager.md]:
```suggestion
[`Sessions`][sessions] library to obtain a random `Session` and forward the `T` message to it.
```

Comment (aesedepece) [reply docs/architecture/managers/sessions-manager.md]: Shall we mention here what happens to the return value of the delegated call?

File: docs/architecture/managers/sessions-manager.md [hunk 6]
```diff
@@ -103,17 +104,21 @@ session_manager_addr
...
+| Message               | Destination           | Input type    | Output type                       | Description                                           |
+|-----------------------|-----------------------|---------------|-----------------------------------|-------------------------------------------------------|
+| GetServerAddress      | ConfigManager         | `()`          | `Option<SocketAddr>`              | Request the config server address                     |
+| GetConnLimits         | ConfigManager         | `()`          | `Option<(u16, u16)>`              | Request the config connections limits                 |
+| GetRandomPeer         | PeersManager          | `()`          | `PeersResult<Option<SocketAddr>>` | Request the address of a peer                         |
+| OutboundTcpConnect    | ConnectionsManager    | `SocketAddr`  | `()`                              | Request a TCP conn to an address                      | 
+| Anycast<GetPeers>     | SessionsManager       | `()`          | `()`                              | Request to send a GetPeers message to a random Session|
```
Comment (aesedepece) [inline docs/architecture/managers/sessions-manager.md]: Please add backticks (``) around the message names and destinations

File: docs/architecture/managers/sessions-manager.md [hunk 7]
```diff
@@ -103,17 +104,21 @@ session_manager_addr
...
-| GetRandomPeer         | PeersManager          | `()`          | `PeersResult<Option<SocketAddr>>` | Request the address of a peer         |
-| OutboundTcpConnect    | ConnectionsManager    | `SocketAddr`  | `()`                              | Request a TCP conn to an address      | 
+| Message               | Destination           | Input type    | Output type                       | Description                                           |
+|-----------------------|-----------------------|---------------|-----------------------------------|-------------------------------------------------------|
+| GetServerAddress      | ConfigManager         | `()`          | `Option<SocketAddr>`              | Request the config server address                     |
+| GetConnLimits         | ConfigManager         | `()`          | `Option<(u16, u16)>`              | Request the config connections limits                 |
+| GetRandomPeer         | PeersManager          | `()`          | `PeersResult<Option<SocketAddr>>` | Request the address of a peer                         |
```
Comment (aesedepece) [inline docs/architecture/managers/sessions-manager.md]:
```suggestion
| GetRandomPeer         | PeersManager          | `()`          | `PeersResult<Option<SocketAddr>>` | Request a random peer address                        |
```

File: docs/architecture/managers/sessions-manager.md (hunk 6 above)
Comment (aesedepece) [inline docs/architecture/managers/sessions-manager.md]:
```suggestion
| `Anycast<GetPeers>`     | `SessionsManager`       | `()`          | `()`                              | Request to forward a GetPeers message to one randomly selected `Session`|
```

File: docs/architecture/managers/sessions-manager.md [hunk 8]
```diff
@@ -169,6 +174,14 @@ connections and try to create a new one.
 
 For further information, see [`ConnectionsManager`][connections_manager].
 
+#### Anycast<GetPeers>
+This message is periodically sent to itself ([`SessionsManager`][sessions_manager]).
```
Comment (aesedepece) [inline docs/architecture/managers/sessions-manager.md]: What for is this sent to itself? Please explain.

File: docs/architecture/managers/sessions-manager.md [hunk 9]
```diff
@@ -169,6 +174,14 @@ connections and try to create a new one.
...
+
+It is a best effort message, its return value is not processed and the [`SessionsManager`][sessions_manager] 
+actor does not even wait for its response after sending it.
+
+This message request to send a `GetPeers` message to a random Session.
```
Comment (aesedepece) [inline docs/architecture/managers/sessions-manager.md]:
```suggestion
This message causes `SessionManager` to forward a `GetPeers` message to one randomly selected `Session` actor. 
```

File: docs/architecture/session.md [hunk 10]
```diff
@@ -28,6 +28,17 @@ Session::create(move |ctx| {
...
+### Incoming: Others -> Peers Manager
+
+These are the messages supported by the peers manager handlers:
+
+| Message          | Input type            | Output type                       | Description            |
+| ---------------- | --------------------- | --------------------------------- | ---------------------- |
+| `GetPeers`       | `()`                  | `()`                              | Empty                  |
+
+#### GetPeers
+The handler of `GetPeers` message is actually empty
```
Comment (aesedepece) [inline docs/architecture/session.md]:
```suggestion
The handler of `GetPeers` message is currently empty.
// TODO Update documentation when `GetPeers` gets any actual functionality.
```

File: docs/architecture/managers/sessions-manager.md (hunk 5 above)
Comment (lrubiorod) [reply docs/architecture/managers/sessions-manager.md]: Added rust code with the function where it happens

File: docs/architecture/managers/sessions-manager.md (hunk 8 above)
Comment (lrubiorod) [reply docs/architecture/managers/sessions-manager.md]: Done!

**Pull request merged**